tests/builtins/functions/test_update_state.py

```diff
@@ -44,10 +51,58 @@ def test_empty_event():
         run_function("update-state", {})
 
 
-def test_success(event, statedb):
+@pytest.fixture
+def workflow_event_topic(sns):
+    return sns.create_topic(Name="app-cirrus-workflow-event")["TopicArn"]
+
+
+@pytest.mark.parametrize(
+    "wf_event_enabled,sfn_state", product((True, False), SfnStatus._member_names_)
+)
+def test_workflow_event_notification(
+    event, statedb, workflow_event_topic, wf_event_enabled, sfn_state, monkeypatch
+):
+    """This tests that workflow events are properly published (if `wf_event_enabled`),
+    with the associated StepFunctions State.  It also verifies that the record is added
+    to the `statedb`.  The content of that record is validated in separate
+    `test_{StateEnum}` tests.
+
+    N.B. the tests end with `wf_event_enabled=False`, such that the loaded
+    lambda_function module does not have the SNS publishing enabled, and the other tests
+    do not require the `workflow_event_topic` fixture to create the topic (as it will
+    not be used).
+    """
+    if wf_event_enabled:
+        expected_msg_count = 1
+        monkeypatch.setenv("CIRRUS_WORKFLOW_EVENT_TOPIC_ARN", workflow_event_topic)
+    else:
+        expected_msg_count = 0
+        monkeypatch.delenv("CIRRUS_WORKFLOW_EVENT_TOPIC_ARN", raising=False)
+    if "cirrus.builtins.functions.update-state.lambda_function" in sys.modules:
+        importlib.reload(
+            sys.modules["cirrus.builtins.functions.update-state.lambda_function"]
+        )
+
+    event["detail"]["status"] = sfn_state
     run_function("update-state", event)
 
     items = statedb.get_dbitems(payload_ids=[EVENT_PAYLOAD_ID])
+    assert len(items) == 1
+    assert "state_updated" in items[0]
```

Comment:
this test is more concerned with the workflow event topic functioning, but I wanted to confirm that enabled or disabled, the state database is updated (and the state database starts out each test invocation empty, so 1 item means it did update).